community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/codegen/LogicalPlanConverter.scala

```diff
@@ -74,7 +74,7 @@ object LogicalPlanConverter {
                                                (e: ast.Expression) => ExpressionConverter.createProjection(e)(context))
       val vars = columns.map {
         case (name, expr) =>
-          val variable = Variable(context.namer.newVarName(), expr.cypherType(context), expr.nullable(context))
+          val variable = Variable(context.namer.newVarName(), symbols.CTAny, expr.nullable(context))
```

Comment:
When we are projecting we are basically erasing the type information, since we are doing

```
Object res = ....
row.set("foo", res);
```
I think we should improve this an keep the type information, but maybe we can do this as a separate piece of work